### DISCUSSION TITLE
最近几个版本的编译更新后重启路由都会恢复初始设置

### DISCUSSION THREAD

**Discussion by suizhenmin** [coolsnowwolf/lede#11823] [Bugs / 问题反馈]
最近几个版本的编译更新后重启路由都会恢复初始设置,请问要 怎么解决这个问题

**Comment by WYC-2020** [comment]
怕不是空间不够了吧

**Reply by suizhenmin** [reply]
16G 的傲腾做的系统盘，不应该空间不够吧，2个区分别给了256和512

**Reply by sss132919** [reply]
说的可能是固件空间满了，而不是你硬盘满了。看看系统-软件包哪里显示的

**Reply by GreatMichaelLee** [reply]
我猜是跟那个改bios 1M分区的改动有关，哈哈，幸亏我有备份

**Comment by suizhenmin** [comment]
查看过来，可用空间剩余98%

**Reply by suizhenmin** [reply]
幸好我也有备份

**Comment by ohcocacola** [comment]
@suizhenmin 同样是这个更新后重置问题…软件包显示剩余700多MB，有解决吗？分享一下，谢谢！

**Reply by suizhenmin** [reply]
没有，我今天又编译更新了一次固件，还是会重置

**Reply by wybb** [reply]
现在有解决的办法了吗？

**Comment by xingshu1990** [comment]
先看下是否是文件系统是只读模式。
检查当前挂载情况： 你可以通过以下命令查看文件系统的挂载状态：
```
mount | grep ' / '
```
将文件系统挂载为读写模式： 如果发现 /（根目录）被挂载为只读，执行以下命令将其挂载为读写：
```
mount -o remount,rw /
```
确认挂载成功： 再次检查挂载状态，确保文件系统已经变为读写模式：
```
mount | grep ' / '
```

有时候重启以后也还是要重新输入：mount -o remount,rw /


**以上是我近期碰到新刷固件不能保存并应用的时候，所做的尝试。如果不是这问题，且不能解决，就等高手来解答**

**Reply by wybb** [reply]
既然可以还原备份文件，那肯定不是读写问题
root@OpenWrt:~# mount | grep ' / '
overlayfs:/overlay on / type overlay (rw,noatime,lowerdir=/,upperdir=/overlay/upper,workdir=/overlay/work,xino=off)

**Reply by xingshu1990** [reply]
> 既然可以还原备份文件，那肯定不是读写问题 root@OpenWrt:~# mount | grep ' / ' overlayfs:/overlay on / type overlay (rw,noatime,lowerdir=/,upperdir=/overlay/upper,workdir=/overlay/work,xino=off)

这样的话，也有可能是基于lede编译的固件里的软件，和其他第三方软件源里的软件冲突了。
见：https://github.com/kenzok8/small/issues/148，
解决办法：双仓库一键命令。

我这边多次编译固件的时候，碰到这种情况。
后来用了上面的双仓库方法，疑似解决。
后来编译 且在使用的固件，除了每次重启后，修改文件不能保存，得手动执行下mount -o remount,rw /
其他都还正常。

**Reply by sss132919** [reply]
这个问题怎么还没有关闭呢？
基本上都是因为存储被占满了啊。。
解决方案就是自己编译，然后存储设定大一点，我都设定1024/2048，再也没遇到过这个问题。

**Reply by wybb** [reply]
> 这个问题怎么还没有关闭呢？ 基本上都是因为存储被占满了啊。。 解决方案就是自己编译，然后存储设定大一点，我都设定1024/2048，再也没遇到过这个问题。

kernel分区1024  root分区2048？

**Reply by sss132919** [reply]
> > 这个问题怎么还没有关闭呢？ 基本上都是因为存储被占满了啊。。 解决方案就是自己编译，然后存储设定大一点，我都设定1024/2048，再也没遇到过这个问题。
> 
> kernel分区1024  root分区2048？

是啊。直到现在都没搞明白，为啥那些发布x86的固件的，存储要搞得那么死，难道人人用傲腾吗？1024+2048也就3g，谁家x86机器系统盘缺这3g。。。

**Reply by wybb** [reply]
> kernel分区

很感谢，经过测试是kernel分区大小的问题，默认16M，设置大一些就可以了
